### ISSUE TITLE
Improve handling images (deleting / sharing etc)

### ISSUE THREAD

**Issue by feutl** [nextcloud/android#4977]
My use case:
I upload all pictures made on my smart phone to nextcloud automatically and move them to the app folder locally. This allows me view the pictures in the nextcloud app immediately without downloading them before.

No I like to sort through my pictures when commuting, deleting the unnecessary ones. Using the gallery app on the phone, I would only delete the local pictures, therefor I use the nextcloud app, but deleting (or moving) pictures is quite some taps away.
It works as expected, I do have my pictures deleted on the NC instance and locally, but I would like to have a much faster way of handling this.

First thing would be, instead of the 3-dots on the right top corner, this actions can be placed directly on the bottom of the screen.  More or less like the Simple Gallery is doing it (https://github.com/SimpleMobileTools/Simple-Gallery) this would allow me to delete or move or even share pictures (or perhaps other files as well) quite easily.

The next thing, which would be great, is to have to skip the "delete confirmation" dialog for this session (once again like simple gallery).

After deleting, move on to the next picture.

Basically this would allow me to have my pictures properly sorted and synced between the device I was shooting it (or synced it to).

I am sure others would benefit from this as well, just thinking about my parents, they delete pictures locally, but this is not reflected on the NC instance right now, but with this process I could tell the, to do so within the nextcloud app quite easily.

**Comment by AndyScherzinger** [comment]
@feutl are you aware of the long-press action which leads to a muli-select mode where you can select multiple files and then hit delete (with the confimration dialog). So you would only have to do the confirmation once.

**Comment by feutl** [comment]
@AndyScherzinger 
Yes I know this, but if you sort out pictures, the list view and also the "Rasteransicht" are not very suitable to sort pictures, especially panoramas ;)
But I think this could be easily implemented (even though I am not a programmer) perhaps code sharing with simple tools code could be an option, but this would make the nextcloud client a little more usable for "home users" and I have 5 of them struggling with pictures, which is the main reason they use nextcloud :) (and not google photos)

**Comment by AndyScherzinger** [comment]
Hi @feutl,

sorry for coming back that late, by bottom actions do you mean: https://github.com/SimpleMobileTools/Simple-Gallery/blob/master/fastlane/metadata/android/en-US/images/phoneScreenshots/app_3.jpg ❓

**Comment by feutl** [comment]
@AndyScherzinger 
don't be sorry, good things do need their time :)
But yes this is what I was referring too. These buttons are very ease to access, the delete buttons also asks if you really want to delete the pictures and you can even disable this prompt on a per session basis. This allows me to go over the pictures quite fast. One after one, delete the ones I don't like and the next one shows up, if I like it I do swipe to the next and so on.

![confirmation](https://user-images.githubusercontent.com/8522012/79755882-169d0900-831a-11ea-8792-3c3bf9be93c5.jpeg)

But you could simply install the app by yourself, you gonna love its simplicity :) perhaps a "deeper" integration would make sense ;)

Something like done with DavX or OnlyOffice etc.

**Comment by AndyScherzinger** [comment]
@nextcloud/designers @jancborchardt what do you think? As for the "quick actions" in the image preview view, like  https://github.com/SimpleMobileTools/Simple-Gallery/blob/master/fastlane/metadata/android/en-US/images/phoneScreenshots/app_3.jpg seems fine to me and would be a good thing.

As for the "don't ask for this session" logic, that might need some more specification how and for which actions this might work, see https://github.com/nextcloud/android/issues/4977#issuecomment-616546850

**Comment by AndyScherzinger** [comment]
cc @tobiasKaminsky info loop

**Comment by tobiasKaminsky** [comment]
I would go for tiny steps and decide which  quick actions we want to have and then implement them…
favorite: yes
edit: not possible
share: yes
delete: yes

**Comment by saywebsolutions** [comment]
very good feedback from @feutl ; currently the workflow to review photos and delete duplicates / unwanted versions of a given shot, is very tiresome. After deleting each photo the user is taken back to the thumbnail overview which really kills flow. And the delete operation should be able to be accomplished in 1 click (not hidden in a context menu), otherwise it slows the workflow down significantly and unnecessarily

**Comment by matzebond** [comment]
@AndyScherzinger
I am also annoyed by the fact that deleting an image throws me back to the grid view and I feel like it should be an easy change to just go to the next picture in the fullscreen view instead (am I right on that?).
I would like to help out and it would be nice if someone cod point me to the right files.

Would you accept such a PR as the new default or would an option in the settings (even more stuff to get my head around :upside_down:) also be needed?

**Comment by tobiasKaminsky** [comment]
Your idea is nice.
I am unsure if / how much work it is, depending on your Android skills.
Of course this PR would be the default hen, there is no need for settings, we aim to have it as easy as possible.

Delete files dialog is called in PreviewImageFragment: onOptionsItemSelected

The result is then handled here: 
https://github.com/nextcloud/android/blob/b74be97a16442d94d0f6d2eafc55ac0de0787b2d/src/main/java/com/owncloud/android/ui/preview/PreviewImageActivity.java#L222-L229

So in first step it is enough to remove the first check and remove "finish()".
Then you would need to switch the PreviewImagePagerAdapter to next image and prevent that user can go back to deleted image.

When deleting the very last image, I would say that UI still should go back to grid view.

If you need any help, just ping me.
I am happy that you want to help, and welcome 🎉

**Comment by matzebond** [comment]
@tobiasKaminsky

Ok first try was to replace
`finish();`
by
```
if (mSavedPosition < mPreviewImagePagerAdapter.getCount() - 1) {
    mSavedPosition += 1;
    mViewPager.setCurrentItem(mSavedPosition);
}
else {
    mSavedPosition = 0;
    finish();
}
```
When swiping back the Fragment with the removed Image is still there and throws an error (filepath does not exist any more) upon getting focused. 
I tried something like 
```
mImageFiles.remove(position);
notifyDataSetChanged();
```
in PreviewImagePagerAdapter but that does not help.
Any idea how to get references to the obsolete Fragments and delete them?

**Comment by tobiasKaminsky** [comment]
It seems not to be that easy: https://stackoverflow.com/questions/25928732/removing-fragments-from-fragmentstatepageradapter

**Comment by tobiasKaminsky** [comment]
Hm returning POSITION_NONE in getItemPosition seems to help.
But this then means that every fragment is re-created every time, which makes the whole purpose of the fragment state pager obsolete :S
